test/testcafe/spec/EnrollAuthenticatorPasswordView_spec.js

```diff
@@ -84,3 +88,18 @@ test('should have the correct reqiurements', async t => {
   await t.expect(enrollPasswordPage.getRequirements()).contains('Your password cannot be any of your last 4 passwords');
   await t.expect(enrollPasswordPage.getRequirements()).contains('A lowercase letter');
 });
+
+test('should call processCreds when the user submits the form', async t => {
+  const enrollPasswordPage = await setup(t);
+  await rerenderWidget({
+    processCreds: ({ username, password }) => {
+      // eslint-disable-next-line no-console
+      console.log(`processCreds called with ${username} and ${password}`);
```

Comment:
what about when processCreds is a promise

Reply:
I modified one of these tests to use the `processCreds(creds, callback)` function signature instead, I am guessing that's what you were referring to. All the promise stuff happens in `this.settings.processCreds`, not the`this.settings.get('processCreds')` that the user supplies.

Reply:
Right, like a test case supporting this example from the [readme](https://github.com/okta/okta-signin-widget#username-and-password):
```
// Passed a creds object {username, password} and a callback for further processing
processCreds: function (creds, callback) {
  // This example demonstrates optional legacy form-based login
  $.ajax({
    method: "POST",
    url: "/logintype",
    data: {
      username : creds.username
    },
    success: function (logintype) {
      if (logintype == "LEGACY") {
        $('#legacyUser').val(creds.username);
        $('#legacyPassword').val(creds.password);
        $('#legacyLogonForm').submit();
      } else {
        callback();
      }
    }
  });
}
```